contrib/sarplus/python/pysarplus/SARPlus.py

```diff
@@ -0,0 +1,362 @@
+"""
+This is the one and only (to rule them all) implementation of SAR.
+"""
+
+import logging
+import pyspark.sql.functions as F
+import pandas as pd
+from pyspark.sql.types import StringType, DoubleType, StructType, StructField, IntegerType, FloatType
+from pyspark.sql.functions import pandas_udf, PandasUDFType
+from pysarplus import SARModel
+
+SIM_COOCCUR = "cooccurrence"
+SIM_JACCARD = "jaccard"
+SIM_LIFT = "lift"
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('sarplus')
+
+class SARPlus:
```

Comment:
would be preferable to match the method signatures used in the python implementation:
reco_utils/recommender/sar/sar_singlenode.py, aside from unique variables for sarplus

for instance many of the fit variables were moved into the init method to make it more consistent with spark ml lib format

Reply:
done